src/utils/fritekstUtils.ts

```diff
@@ -1,153 +1,175 @@
+};
 
-        return fritekstTabellKopi.map((fritekstTabellElement) => {
+export const oppdaterFritekstTabellMedTekst = (
+    fritekstTabellKopi: string[][],
+    nyFritekst: string
+): { nyFritekstTabell: string[][]; flettefeltNummer?: number; nyttFritekstElement?: string } => {
+    const nyFritekstSterilisert = fjernSpesialKarakterer(nyFritekst);
+    const gammelFritekst = dobbelTabellTilStreng(fritekstTabellKopi);
+    const antallTegnLagtTil = finnAntallTegnLagtTil(nyFritekstSterilisert, gammelFritekst);
+    const endringsIndeks = finnEndringsIndeks(nyFritekstSterilisert, gammelFritekst);
+
+    let karakterTeller = 0;
+    let erEndringGjort = false;
+
+    let nyttFritekstElement;
+    let flettefeltIndeksITabell = 0;
+    let nyFritekstTabell = fritekstTabellKopi;
+    if (endringsIndeks !== -1) {
+        nyFritekstTabell = fritekstTabellKopi.map((fritekstTabellElement) => {
             const gammelKarakterTeller = karakterTeller;
             karakterTeller += finnKaraktererIListeMedStrenger(fritekstTabellElement);
 
             if (karakterTeller > endringsIndeks && !erEndringGjort) {
                 erEndringGjort = true;
-                const { tabellIndeks, nyttFritekstElement } = finnTabellIndeksOgNyttFritekstElement(
+                const tabbellIndeksOgNyttFritekstElement = finnTabellIndeksOgNyttFritekstElement(
                     endringsIndeks,
                     nyFritekstSterilisert,
                     fritekstTabellElement,
                     antallTegnLagtTil,
                     gammelKarakterTeller
                 );
-
-
-                fritekstTabellElement[tabellIndeks] = nyttFritekstElement; 
-				
+                const tabellIndeks = tabbellIndeksOgNyttFritekstElement.tabellIndeks;
+                nyttFritekstElement = tabbellIndeksOgNyttFritekstElement.nyttFritekstElement;
+                fritekstTabellElement[tabellIndeks] = nyttFritekstElement;
+                flettefeltIndeksITabell += tabellIndeks;
                 return fritekstTabellElement;
             } else {
+                if (!erEndringGjort) {
+                    flettefeltIndeksITabell += fritekstTabellElement.length - 1;
+                }
                 return fritekstTabellElement;
             }
         });
-}
-
-export const oppdaterFritekstTabellMedDropdown = (fritekstTabellKopi: string[][], innholdIndeks: number, delseksjon: SanityDelseksjon, valgIndeks: number): string[][] => {
-	fritekstTabellKopi[innholdIndeks] = sanityBlocktekstToHtml(
-		(delseksjon.innhold[innholdIndeks] as SanityDropdown).valg[valgIndeks]
-	);
-	return fritekstTabellKopi;
-}
-
-export const oppdaterFritekstTabellMedFlettefelt = (fritekstTabellKopi: string[][], innholdIndeks: number, flettefeltIndeks: number, nyVerdi: string): string[][] => {
-        fritekstTabellKopi[innholdIndeks][flettefeltIndeks * 2 + 1] = nyVerdi;
-		return fritekstTabellKopi
-}
+    }
+
+    if (nyttFritekstElement !== undefined && flettefeltIndeksITabell % 2 == 1) {
```

Comment:
Her tenker jeg at man kan legge inn en kommentar som forklarer sjekken. Eventuelt legge en kommentar i `innholdTilflettefeltTabell()`

Reply:
Ryddig 👍

Reply:
Er det denne sjekken som håndterer logikken for å oppdatere flettefelt-tabellen?